### ISSUE TITLE
Creating lbaas pool with persistent type SOURCE_IP

### ISSUE THREAD

**Issue by sirajudheenam** [terraform-providers/terraform-provider-openstack#57]
### Terraform Version
Terraform v0.9.11

### Affected Resource(s)
- openstack_lb_pool_v2 

### Terraform Configuration Files
```hcl

# LOAD BALANCER
resource "openstack_lb_loadbalancer_v2" "lb_1" {
	name = "TEST_LB"
	# Your Private network sub-net ID {source openrc; openstack network list}
  vip_subnet_id = "265fcfa3-eb18-4e2d-9e29-2f8f4506106f"
}

#LISTENER 1
resource "openstack_lb_listener_v2" "listener_1" {
  name = "listener01"
  description = "This is the listener description"
  protocol        = "HTTP"
  protocol_port   = 8080
  loadbalancer_id = "${ openstack_lb_loadbalancer_v2.lb_1.id }"
  region = "${var.region}"
}

#LISTENER 2
resource "openstack_lb_listener_v2" "listener_2" {
  name = "listener02"
  description = "This is the listener 2 description"
  protocol        = "HTTP"
  protocol_port   = 8081
  loadbalancer_id = "${ openstack_lb_loadbalancer_v2.lb_1.id }"
  region = "${var.region}"
}

#POOL 1
resource "openstack_lb_pool_v2" "pool_1" {
  name = "pool01"
  protocol    = "HTTP"
  lb_method   = "ROUND_ROBIN"
  listener_id = "${ openstack_lb_listener_v2.listener_1.id }"
  # loadbalancer_id = "${ openstack_lb_loadbalancer_v2.lb_1.id }"
  region = "${var.region}"

  persistence {
    type        = "APP_COOKIE"
    cookie_name = "Cookie_test"
  }
}

#POOL 2
resource "openstack_lb_pool_v2" "pool_2" {
  name = "pool02"
  protocol    = "HTTP"
  lb_method   = "ROUND_ROBIN"
  listener_id = "${ openstack_lb_listener_v2.listener_2.id }"
  # loadbalancer_id = "${ openstack_lb_loadbalancer_v2.lb_1.id }"
  region = "${var.region}"

  persistence {
    type        = "SOURCE_IP"
    cookie_name = "null"
  }
}
```

### Debug Output

https://gist.github.com/sirajudheenam/a5d80dba18bfde7c4c58a8bf0e744e38

### Panic Output
N/A

### Expected Behavior
It should have created the load balancer pool with persistent type as SOURCE_IP without the need of cookie_name. 
Because I can create with this neutron cli command the same load balancer pool without cookie_name.
`neutron lbaas-pool-create --name "LB_HTTP_POOL" --lb-algorithm ROUND_ROBIN --loadbalancer a55a3fe0-3160-4a24-a7aa-e7487173fe73 --protocol HTTP --session-persistence type=SOURCE_IP`

### Actual Behavior
It produced error saying that `openstack_lb_pool_v2.pool_2: Error creating OpenStack LBaaSV2 pool: Invalid request due to incorrect syntax or missing required parameters.`

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
OpenStack Version: Mitaka

### References
None.

**Comment by fatmcgav** [comment]
@sirajudheenam Ah, ok... So on the surface, TF is doing what it should, as you've explicitly set 'cookie = null' for `"openstack_lb_pool_v2.pool_2"`. 

However looking at the resource schema, I can see that `cookie_name` is set to required: https://github.com/terraform-providers/terraform-provider-openstack/blob/master/openstack/resource_openstack_lb_pool_v2.go#L115

So I'm presuming you set the value to 'null' in an attempt to satisfy the value being required?

Either way, it's a bug, as `cookie_name` should only be required if `type` is set to to `HTTP_COOKIE` or `APP_COOKIE`.

I'll get a PR up shortly that you can test with if you're happy enough to try compiling from source?

**Comment by sirajudheenam** [comment]
That would be perfect. Thanks.

**Comment by fatmcgav** [comment]
@sirajudheenam An initial fix has been added in #58. Feel free to test that and confirm that it works for you :)

**Comment by sirajudheenam** [comment]
This works for me.
